notes/lecture6.tex

```diff
@@ -0,0 +1,253 @@
+\[
+x_{t+1} = x_t - \eta_t \nabla f(x_t) = x_t - \eta_t \left(Ax_t - b\right)
+\]
+Subtracting $x^*$ from both sides of this equation and using the property that $Ax^* - b = \nabla f(x^*) = 0$:
+\begin{align*}
+x_{t+1} - x^* &= \left(x_t - \eta_t\left(Ax_t - b\right)\right) - \left(x^* - \eta_t\left(Ax^* - b\right)\right) \\
+&= (I - \eta_t A)(x_t - x^*).
+\end{align*}
+Thus:
+\begin{align*}
+\left\|x_{t+1} - x^*\right\|_2 &\leq \left\|(I - \eta_t A)\right\|_2 \left\|(x_t - x^*)\right\|_2 \\
+&\leq \left(\prod_{k=1}^t \left\| I - \eta_k A\right\|_2\right) \left\|x_1 - x^*\right\|_2.
+\end{align*}
+
+Set $\eta_k = \frac{1}{\beta}$ for all $k$. Note that $\frac{\alpha}{\beta}I \preceq \frac{1}{\beta}A \preceq I$, so:
+\begin{equation*}
+\max_{A \in \R^{n\times n}} \left\|I - \frac{1}{\beta} A\right \|_2 = 1 - \frac{\alpha}{\beta} = 1 - \frac{1}{\kappa}.
+\end{equation*}
+It follows that:
+\begin{align*}
+\left\|x_{t+1} - x^*\right\|_2 &\leq \left(1 - \frac{1}{\kappa}\right)^t \left\|x_1 - x^*\right\|_2 \\
+&\leq \exp \left(-\frac{t}{\kappa}\right) \left\|x_1 - x^*\right\|_2.
+\end{align*}
+\end{proof}
+
+\subsection{Accelerated gradient descent}
+In the previous section, we proved a lower bound on the convergence rate. In this section, we would like to improve on this. Consider whether there was any point where we were careless in the proof above? One obvious candidate is that our choice of step size, $\eta_k = \frac{1}{\beta}$, was chosen rather arbitrarily. In fact, by choosing the sequence $\eta_k$ we can select \textit{any} degree-$t$ polynomial of the form:
+\[
+p(A) = \prod_{k=1}^t \left(I - \eta_K A\right).
+\]
+
+Note that:
+% TODO: this is apparently an equality, do we want to strengthen it?
+\begin{equation*}
+\left\|p(A)\right\|_2 \leq \max_{x \in \lambda(A)} \left|p(x)\right|
+\end{equation*}
+% Note this is actually also an equality.
+where $p(A)$ is a matrix polynomial, and $p(t)$ is the corresponding scalar polynomial. In general, we may not know the set of eigenvalues $\lambda(A)$, but we do know that all eigenvalues are in the range $[\alpha, \beta]$. Relaxing the inequality accordingly:
+\begin{equation*}
+\left\|p(A)\right\| \leq \max_{x \in [\alpha, \beta]} \left|p(x)\right|.
+\end{equation*}
+
+We want a polynomial $p(t)$ that takes on small values in $[\alpha, \beta]$. We impose an additional normalization constraint that $p(0) = 1$ (otherwise we could `cheat' by scaling the polynomial down everywhere on its domain), but allow it to take on arbitrary values elsewhere.
+
+\subsubsection{A naive polynomial solution}
+A naive solution chooses a uniform step size $\eta_t = \frac{2}{\alpha + \beta}$. Note that:
+\[
+\max_{x \in [\alpha, \beta]} \left|1 - \frac{2}{\alpha + \beta} x\right| = \frac{\beta - \alpha}{\alpha + \beta} \leq \frac{\beta}{\alpha} = \kappa,
+\]
+recovering the same convergence rate we proved previously.
+\begin{figure}[h]
+\includegraphics[width=15cm]{figures/lecture6-naive_polynome.pdf}
```

Comment:
please specify width as a fraction of `\textwidth`. E.g., `0.9\textwidth` if that's what you want.